lib/run_part.c

```diff
@@ -51,68 +52,78 @@ int run_parts(const char *directory, const char *name, const char *action)
 	int n;
 	int execute_result = 0;
 
-	scanlist = scandir(directory, &namelist, 0, alphasort);
+	int dfd = open (directory, O_PATH | O_DIRECTORY | O_CLOEXEC);
+	if (dfd == -1) {
+		fprintf(shadow_logfd, "failed open directory %s: %s\n", directory, strerror(errno));
+		return (1);
+	}
+
+	scanlist = scandir (directory, &namelist, 0, alphasort);
```

Comment:
There's scandirat(3).  We should probably also use that.

Reply:
Using `scandirat(3)` when available.